app/components/Form/ProjectManagerForm.tsx

```diff
@@ -38,27 +69,168 @@ const ProjecManagerForm: React.ForwardRefRenderFunction<any, Props> = (
+    };
+
+    // If a form_change already exists, and the payload contains a cifUserId update it
+    if (formChangeId && change.cifUserId) {
+      applyUpdateFormChangeMutation({
+        variables: {
+          input: {
+            id: formChangeId,
+            formChangePatch: {
+              newFormData: data,
+            },
+          },
+        },
+        optimisticResponse: {
+          updateFormChange: {
+            formChange: {
+              id: formChangeId,
+              newFormData: data,
+            },
+          },
+        },
+        debounceKey: formChangeId,
+      });
+      // If a form_change does not exist, and the payload contains a cifUserId create a form_change record
+    } else if (change.cifUserId) {
+      addManager(data);
+    }
+    // If a form_change exists, and the payload does not contain a cifUserId delete it
+    else if (formChangeId && !change.cifUserId) {
+      deleteManager(formChangeId);
+    }
+  };
+
+  props.setValidatingForm({
+    selfValidate: () => {
+      return Object.keys(formRefs.current).reduce((agg, formId) => {
+        const formObject = formRefs.current[formId];
+        return [...agg, ...validateFormWithErrors(formObject)];
+      }, []);
+    },
+  });
+
+  return (
+    <>
+      <Grid cols={10} align="center">
+        <Grid.Row>
+          <Grid.Col span={10}>
+            <FormBorder title="Project Managers">
+              {projectRevision.managerFormChanges.edges.map(({ node }) => (
+                <React.Fragment key={node.projectManagerLabel.id}>
+                  <Grid.Row>
+                    <label>{node.projectManagerLabel.label}</label>
+                  </Grid.Row>
+                  <Grid.Row>
+                    <Grid.Col span={6}>
+                      <FormBase
+                        id={`form-manager-${node.projectManagerLabel.label}`}
+                        idPrefix={`form-${node.projectManagerLabel.id}`}
+                        ref={(el) =>
+                          (formRefs.current[node.projectManagerLabel.id] = el)
+                        }
+                        formData={node.formChange?.newFormData}
+                        onChange={(change) => {
+                          createOrUpdateFormChange(
+                            node.formChange?.id,
+                            node.projectManagerLabel.rowId,
+                            change.formData
+                          );
+                        }}
+                        schema={managerSchema}
+                        uiSchema={uiSchema}
+                        ObjectFieldTemplate={EmptyObjectFieldTemplate}
+                      />
+                    </Grid.Col>
+                    <Grid.Col span={4}>
+                      <Button
+                        variant="secondary"
+                        size="small"
+                        onClick={() => deleteManager(node.formChange?.id)}
+                      >
+                        Clear
```

Comment:
should be disabled if there is no manager selected, or if the mutation is already in flight for this manager. You may need to refactor the react fragment into a separate component to track individual mutations

Reply:
We could also make the clear button a noop if the mutation is in flight or if there is no manger selectd, I don't think that would harm the UX, things would be cleared again from the user's perspective